azure-iot-hub/azure/iot/hub/iothub_amqp_client.py

```diff
@@ -45,13 +52,36 @@ def _build_amqp_endpoint(self, hostname, shared_access_key_name, shared_access_k
         endpoint = endpoint + ":{}@{}".format(quote_plus(sas_token), hostname)
         return endpoint
 
-    def __init__(self, hostname, shared_access_key_name, shared_access_key):
-        self.endpoint = self._build_amqp_endpoint(
-            hostname, shared_access_key_name, shared_access_key
-        )
-        operation = "/messages/devicebound"
-        target = "amqps://" + self.endpoint + operation
-        self.amqp_client = uamqp.SendClient(target)
+    def __init__(
+        self,
+        hostname,
+        shared_access_key_name=None,
+        shared_access_key=None,
+        token_credential=None,
+        token_scope="https://iothubs.azure.net/.default",
+    ):
```

Comment:
The way you have these parameters and the mutual exclusion between them, you should consider making a base class with 2 child classes -- like `IoTHubAmqpClientBase` with `IoTHubAmqpClientConnectionString` and `IoTHubAmqpClientTokenAuth` or something like that.

I don't think you _must_ change it, but using 2 derived classes just feels a little warmer and a little fuzzier to me.

Reply:
Yeah, that does sound better. I'll look into it.

Reply:
Done